### PR TITLE
FEAT: Add Footer

### PR DESCRIPTION
**Pull request by naumch1k**
## Description
This PR introduces page footer.

## Task Link
[Project Figma](https://www.figma.com/design/YKZFEj97V6fe8LADQR5lcq/ONE-FALL?node-id=0-1&t=Gr51OfE2P1Uth1KZ-0)

## Changes Made
- Updated `useNavigation` hook to hide the main navigation when the user scrolls to the footer.
- Implemented `useFormWithValidation` to prevent empty or multiple form submissions.
- Added `<Loader />` UI component for reuse across the webpage.
- Added constants re-export to shorten multiple file imports.

### Responsiveness
desktop
tablet
mobile

### Form Validation
Form uses built-in HTML validation to check user input. Error messages are shown until the input meets the required criteria. When user submits form, here is a loader to indicate that form is being sent, while submit button is disabled to prevent multiple submissions.

If submission is successful, the form resets, and button text changes to 'MESSAGE SENT ✔'. After 10 seconds, button text returns to 'SEND MESSAGE'.

https://github.com/user-attachments/assets/bcfcabf7-af8f-4f94-a3f6-a77ab7c8d7de

## Usage
Form still needs to be integrated with email sending service, for now it is set up using a little timeout to simulate form being sent.

```tsx
  const simulateFormSubmission = () => {
    return new Promise<void>(resolve => {
      setTimeout(() => resolve(), 2000)
    })
  }

  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    setIsSubmitting(true)

    simulateFormSubmission()
      .then(() => {
        setFormSuccessfullySent(true)
        resetForm()
      })
      .catch(error => console.log(error))
      .finally(() => setIsSubmitting(false))
  }
```

### CODE REVIEW

**Review by b3dyk — CHANGES_REQUESTED** [review]

File: src/components/ui/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { TextareaHTMLAttributes } from 'react'
+import styles from './TextArea.module.css'
+
+interface ITextAreaProps extends TextareaHTMLAttributes<HTMLTextAreaElement> {
+  error: boolean
+  errorMessage: string
+}
+
+export const TextArea = ({
```
[the remark this answers is not in the document]
Comment (naumch1k) [reply src/components/ui/TextArea/TextArea.tsx:9]: Hey @b3dyk , I took care of the scrollbar styling. Could you please take a quick look?

File: src/components/ui/TextArea/TextArea.module.css [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  font-size: 16px;
+  line-height: 1.2;
+  color: rgb(var(--rgb-foreground-secondary));
+  background-color: transparent;
+  border: 1px solid rgb(var(--rgb-foreground-secondary));
+  border-radius: var(--border-radius);
+  outline: none;
+  resize: vertical;
+  transition: var(--common-transition);
+}
```
Comment (b3dyk) [inline src/components/ui/TextArea/TextArea.module.css:15]: Would be nice to add min-height as well. At least 100 px but look through the design and define it

**Review by b3dyk — DISMISSED** [review]

File: src/components/ui/TextArea/TextArea.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (b3dyk) [reply src/components/ui/TextArea/TextArea.tsx:9]: nicely done

File: src/components/ui/TextArea/TextArea.module.css (hunk 2 above)
Comment (naumch1k) [reply src/components/ui/TextArea/TextArea.module.css:15]: @b3dyk Done!

**Pull request merged**